lib/SIL/IR/OperandOwnership.cpp

```diff
@@ -399,20 +408,27 @@ OperandOwnershipClassifier::visitStoreBorrowInst(StoreBorrowInst *i) {
   return OperandOwnership::TrivialUse;
 }
 
-static OperandOwnership getFunctionArgOwnership(SILArgumentConvention argConv) {
+OperandOwnership OperandOwnershipClassifier::getFunctionArgOwnership(
+    SILArgumentConvention argConv, bool hasScopeInCaller) {
+
   switch (argConv) {
   case SILArgumentConvention::Indirect_In:
   case SILArgumentConvention::Direct_Owned:
     return OperandOwnership::ForwardingConsume;
 
-  // A guaranteed argument is forwarded into the callee. However, from the
+  // A guaranteed argument is forwarded into the callee. If the call itself has
+  // no scope (i.e. it is a single apply, try_apply or yield), then from the
   // caller's point of view it looks like an instantaneous use. Consequently,
   // owned values may be passed to guaranteed arguments without an explicit
-  // borrow scope in the caller.
+  // borrow scope in the caller. In contrast, a begin_apply /does/ have an
+  // explicit borrow scope in the caller so we must treat arguments passed to it
+  // as being borrowed for the entire borrow scope.
```

Comment:
`for the entire borrow scope` -> `for the entire region of the coroutines execution that makes up that borrow scope`.

Or something like that.